### DISCUSSION TITLE
2011-11-29 Core team update

### DISCUSSION THREAD

**Discussion by mm-near** [near/community#140] [Core]
# Highlights:
* validation selection algorithm launched on testnet

# TL;DR
@pmnoxx - improving network code & documentation, network performance / network rate-limiting / load-testing
@Longarithm - improving fees / debugging performance inconsistencies
@mzhangmzz -   improve chunk fetching / improving block validation error handling
@yordanmadzhunkov - fuzz-testing 
@EgorKulikov - working on different validator assignment (for 'more validators' on Nov 29)
@abacabadabacaba - better storage analysis (replace Merkle tree with AVL)

# Done:
@pmnoxx - even more PRs merged (around 40 PRs) with documentation improvements
@pmnoxx - implemented load tester (and fixing issues that it discovered)
@pmnoxx - improving our test framework (make tests work from any folder)
@pmnoxx - 990% of changes required to protect nodes from spamming

@Longarithm - fixed vulnerabilities in contract preparation 
@Longarithm - presentation with options on how to solve fee issues
@Longarithm - fixed transaction hash collision issue

@mzhangmzz - finished request to fetch chunks in advance, but it caused some issues (near/nearcore/5360) - need to investigate
@mzhangmzz - Tried (near/nearcore/5144) - but more difficult than expected - so postponed for now.

@EgorKulikov - Stabilised and pushed new validator selection algorithm.

@mm-near - prepared documentation fixit for next week.
@mm-near - understanding and debugging the code around our new validator selection and canary push process

# Progress:

@pmnoxx - many PRs in the queue: adding stats/support for ThrottlerController
@pmnoxx - more documentation
@pmnoxx - improving how we use Time in network codde

@Longarithm - working on config separation (near/nearcore/4447)

@mzhangmzz - debugging why (near/nearcore/5360) caused the issues

@EgorKulikov - review the existing tickets and look at the garbage collection.

**Comment by maxzaver** [comment]
> fixed transaction hash collision issue

@mm-near  What is this referring to? Could you link a PR?

**Reply by mm-near** [reply]
https://github.com/near/nearcore/pull/5482
related to:
https://github.com/near/nearcore/issues/3779
